lvsfunc/comparison.py

```diff
@@ -536,6 +541,23 @@ def tvbd_diff(tv: vs.VideoNode, bd: vs.VideoNode,
     if not frames:
         raise ValueError("tvbd_diff: no differences found")
 
+    if write_log:
+        from itertools import groupby, count
+
+        def intervals(frames: list) -> list:
```

Comment:
no real reason to nest this this deeply, can just make it a top-level function like `_get_diff_intervals`
also the correct typehint for `frames` should be `typing.List[int]` and it returns `typing.List[typing.Tuple[int, int]]`